src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -0,0 +1,71 @@
+package tech.cryptonomic.conseil.metadata
+
+import tech.cryptonomic.conseil.config.MetadataOverridesConfiguration
+import tech.cryptonomic.conseil.config.Platforms.PlatformsConfiguration
+import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.{Attribute, Entity, Network, Platform}
+import tech.cryptonomic.conseil.metadata.Data.{AttributeData, EntityData, NetworkData, PlatformData}
+import tech.cryptonomic.conseil.tezos.TezosPlatformDiscoveryOperations
+import tech.cryptonomic.conseil.util.OptionUtil.when
+import tech.cryptonomic.conseil.util.ConfigUtil
+
+import scala.concurrent.Future.successful
+import scala.concurrent.{ExecutionContext, Future}
+
+class MetadataService(config: PlatformsConfiguration,
+                      tezosPlatformDiscoveryOperations: TezosPlatformDiscoveryOperations,
+                      metadataOverrides: MetadataOverridesConfiguration) {
+
+  def getPlatforms: List[Platform] = {
+    ConfigUtil
+      .getPlatforms(config)
+      .flatMap(platform => PlatformData(PlatformPath(platform.name), platform).apply(metadataOverrides))
+  }
+
+  def getNetworks(path: PlatformPath): Option[List[Network]] = {
+    when(metadataOverrides.isVisible(path)) {
+      ConfigUtil
+        .getNetworks(config, path.platform)
+        .flatMap(network => NetworkData(path.addLevel(network.name), network).apply(metadataOverrides))
+    }
+  }
+
+  def getEntities(path: NetworkPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Entity]]] = tezosPlatformDiscoveryOperations
+    .getEntities
+    .map(entities => {
+      when(exists(path) && metadataOverrides.isVisible(path)) {
+        entities
+          .flatMap(entity => EntityData(path.addLevel(entity.name), entity).apply(metadataOverrides))
+      }
+    })
+
+  def getTableAttributes(path: EntityPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] = {
+    tezosPlatformDiscoveryOperations
+      .getTableAttributes(path.entity)
+      .map {
+        _.flatMap {
+          attributes => when(exists(path.up) && metadataOverrides.isVisible(path)) {
+            attributes.flatMap(attribute => AttributeData(path.addLevel(attribute.name), attribute).apply(metadataOverrides))
+          }
+        }
+      }
+  }
+
+  def getAttributeValues(platform: String, network: String, entity: String, attribute: String)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[String]]] = {
+    if (exists(NetworkPath(platform, network)))
+      tezosPlatformDiscoveryOperations.listAttributeValues(entity, attribute).map(Some(_))
+    else
+      successful(None)
+  }
+
+
+  def getAttributesValuesWithFilterRoute(platform: String, network: String, entity: String, attribute: String, filter: String)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[String]]] = {
```

Comment:
Does this needs to be a separate method from `getAttributeValues` ?

Would an Optional param suffice?

Reply:
Nice catch!